src/punycode.cpp

```diff
@@ -22,6 +23,40 @@ static constexpr char digit_to_char(int32_t digit) {
   return digit < 26 ? char(digit + 97) : char(digit + 22);
 }
 
+static constexpr bool begins_with(std::string_view view,
```

Comment:
We now have multiple definitions of begins_with and is_ascii. The final PR should clean that up and have just one definition per function.